src/java/org/apache/cassandra/db/filter/RowFilter.java

```diff
@@ -79,13 +79,16 @@ public class RowFilter
     private static final Logger logger = LoggerFactory.getLogger(RowFilter.class);
 
     public static final Serializer serializer = new Serializer();
-    public static final RowFilter NONE = new RowFilter(FilterElement.NONE);
+    public static final RowFilter NONE = new RowFilter(FilterElement.NONE, false);
 
     private final FilterElement root;
 
-    protected RowFilter(FilterElement root)
+    public final boolean allowFiltering;
```

Comment:
We have a mixed use of `allowFiltering` and `allowsFiltering` around. I would homogenize it, so it's easier to search.

Reply:
`ALLOW FILTERING` has always been kind of a transversal concept while it was a CQL thing. Putting it in the `RowFilter` somehow feels like it finally has a clear home. I think it would be super nice to add some brief JavaDoc here saying what `ALLOW FILTERING` is, for the sake of newcomers.